pandas/_testing/asserters.py

```diff
@@ -672,8 +687,19 @@ def _raise(left, right, err_msg):
         raise AssertionError(err_msg)
 
     # compare shape and values
-    if not array_equivalent(left, right, strict_nan=strict_nan):
-        _raise(left, right, err_msg)
+    if check_exact:
```

Comment:
What tests were affected by this?

Reply:
test_pairwise_with_other.

Reply:
Can you just xfail that test instead of this change?